lib/puppet/provider/ec2_securitygroup/v2.rb

```diff
@@ -23,7 +23,7 @@ def self.instances
   def self.prefetch(resources)
     instances.each do |prov|
       if resource = resources[prov.name] # rubocop:disable Lint/AssignmentInCondition
-        resource.provider = prov if resource[:region] == prov.region
+        resource.provider = prov if (resource[:region] || ENV['AWS_REGION']) == prov.region
```

Comment:
Regarding the two methods of passing region, this mirrors most of the AWS tools which will use the explicitly passed region OR fall back to the value of the Environment variable. It allows for some interesting reuse and in practice is less confusing that in theory in my experience. We should decide on whether we want to follow that convention, and if so agree we should better specify this behaviour and provide examples.

The linked example will only work if you have one puppet run per region, due to the requirement for unique resources in the graph. In the future we should definitely investigate other approaches here, including composite namevars ala https://github.com/puppetlabs/puppetlabs-aws/pull/124. A current pattern is to use the region var to make unique composite names. ie. {region}-{name}.